bindgen/runtime.ml

```diff
@@ -0,0 +1,20 @@
+type lifetime =
+  | Function
+      (** The value can live for the lifetime of the function call, which upon return will signal that the 
+               value can be dropped (finalizer?) *)
+  | Ocaml  (** The value is managed by the OCaml runtime *)
+  | C
+      (** The value is allocated and passed to C which is then in charge of cleaning it up *)
+
+type 'a cptr = { lifetime : lifetime; addr : nativeint }
+
+external bindgen_alloc : size:int -> nativeint = "bindgen_alloc"
+external bindgen_free : nativeint -> unit = "bindgen_free"
+
+let sizeof _ = 4 (* TODO: how to handle different types? *)
+
+let create_ptr (value : 'a) : 'a cptr =
+  let addr = bindgen_alloc ~size:(sizeof value) in
+  print_endline ("Addr: " ^ Nativeint.to_string addr);
+  Gc.finalise bindgen_free addr;
```

Comment:
ocaml garbage collector handles freeing it up when its no longer used. This will need to be customisable hence the `lifetime` field.